src/commands/stepfunctions/awsCommands.ts

```diff
@@ -0,0 +1,316 @@
+import {
+  CloudWatchLogsClient,
+  CreateLogGroupCommand,
+  CreateLogGroupCommandOutput,
+  DeleteSubscriptionFilterCommand,
+  DeleteSubscriptionFilterCommandOutput,
+  DescribeSubscriptionFiltersCommand,
+  DescribeSubscriptionFiltersCommandOutput,
+  PutSubscriptionFilterCommand,
+  PutSubscriptionFilterCommandOutput,
+} from '@aws-sdk/client-cloudwatch-logs'
+import {
+  AttachRolePolicyCommand,
+  AttachRolePolicyCommandOutput,
+  CreatePolicyCommand,
+  CreatePolicyCommandOutput,
+  IAMClient,
+} from '@aws-sdk/client-iam'
+import {
+  DescribeStateMachineCommand,
+  ListTagsForResourceCommand,
+  TagResourceCommand,
+  TagResourceCommandOutput,
+  UntagResourceCommand,
+  UntagResourceCommandOutput,
+  UpdateStateMachineCommand,
+  UpdateStateMachineCommandOutput,
+} from '@aws-sdk/client-sfn'
+import {DescribeStateMachineCommandOutput, ListTagsForResourceCommandOutput, Tag} from '@aws-sdk/client-sfn'
+import {SFNClient} from '@aws-sdk/client-sfn/dist-types/SFNClient'
+import {BaseContext} from 'clipanion'
+
+import {buildLogAccessPolicyName, displayChanges} from './helpers'
+
+export const describeStateMachine = async (
+  stepFunctionsClient: SFNClient,
+  stepFunctionArn: string
+): Promise<DescribeStateMachineCommandOutput> => {
+  const input = {stateMachineArn: stepFunctionArn}
+  const command = new DescribeStateMachineCommand(input)
+  const data = await stepFunctionsClient.send(command)
+
+  return data
+}
+
+export const listTagsForResource = async (
+  stepFunctionsClient: SFNClient,
+  stepFunctionArn: string
+): Promise<ListTagsForResourceCommandOutput> => {
+  const input = {resourceArn: stepFunctionArn}
+  const command = new ListTagsForResourceCommand(input)
+  const data = await stepFunctionsClient.send(command)
+
+  return data
+}
+
+export const putSubscriptionFilter = async (
+  cloudWatchLogsClient: CloudWatchLogsClient,
+  forwarderArn: string,
+  filterName: string,
+  logGroupName: string,
+  stepFunctionArn: string,
+  context: BaseContext,
+  dryRun: boolean
+): Promise<PutSubscriptionFilterCommandOutput | undefined> => {
+  // Running this function multiple times would not create duplicate filters (old filter with the same name would be overwritten).
+  // However, two filters with the same destination forwarder can exist when the filter names are different.
+
+  const input = {
+    destinationArn: forwarderArn,
+    filterName,
+    filterPattern: '',
+    logGroupName,
+  }
+  const command = new PutSubscriptionFilterCommand(input)
+  const commandName = 'PutSubscriptionFilter'
+  displayChanges(stepFunctionArn, context, commandName, dryRun, input)
+  if (!dryRun) {
+    const data = await cloudWatchLogsClient.send(command)
+    // Even if the same filter name is created before, the response is still 200.
+    // there are no way to tell
+    context.stdout.write(
+      `Subscription filter ${filterName} is created or the original filter ${filterName} is overwritten.\nt`
```

Comment:
```suggestion
      `Subscription filter ${filterName} is created or the original filter ${filterName} is overwritten.\n`
```

Reply:
@kimi-p you missed this one I think. Is it expected to have a single `t` at the beginning of the line? (if not, let's fix it in one of the next PRs)